scripts/pre_ingest_sanitisation.py

```diff
@@ -0,0 +1,236 @@
+        if filecmp.cmp(src, dest):
+            print 'but it\'s OK, the files were the same anyway'
+        else:
+            print 'DISASTER - the files were different!\n\n\n'
+    else:
+        shutil.move(src,dest)      
+
+
+def GetFilename(filename_or_path):
+    return str(filename_or_path).split('/')[-1]
+
+if not os.path.exists(dest_root_path): os.mkdir(dest_root_path)
+
+print 'Copying data...'
+sys_call = 'cp -r %s* %s'%(source_path, dest_root_path)
+os.system(sys_call)
+print 'Data copied'
+
+os.mkdir(dest_root_path + 'one_amp')
+os.mkdir(dest_root_path + 'two_amp')
+os.mkdir(dest_root_path + 'wrong_size')
+os.mkdir(dest_root_path + 'bad_files')
+
+original_fileslist = []
+for path, subdirs, files in os.walk(dest_root_path):
+    for name in files:
+        if name.endswith('.fits'):
+            original_fileslist.append(os.path.join(path, name))
+
+
+bad_match = ['fitstest.fits', 'test.fits', 'ut161210.f075.fits']
+pop_list = []
+for filename in original_fileslist:
+    for _ in bad_match:
+        if filename.find(_)!=-1:
+            dest_path = dest_root_path + 'bad_files/'
+            dest_filename = dest_path + '_from_' +filename.replace('/','_') #filenames are not unique
+            SafeMove(filename, dest_filename)
+            pop_list.append(filename)
+            print "Moved: %s to \n%s"%(filename, dest_filename)
+            break # don't want to match twice per file
+            
+for popname in pop_list:
+    original_fileslist.remove(popname)
+
+
+headers = {}
+
+nfiles = len(original_fileslist)
+for i, filename in enumerate(original_fileslist):
+    hdulist = pf.open(filename)
+    main_header = hdulist[0].header
+    headers[filename] = main_header
+    if i%500==0: print 'loaded %s of %s'%(i, nfiles)
+
+print 'Finished loading headers'
+
+
+counter = 0
+for filename in headers.keys():
+    if headers[filename]['NAMPSYX']=='1 1':
+        SafeMove(filename, dest_root_path + 'one_amp/' + GetFilename(filename))
+        counter += 1
+        headers.pop(filename)
+print 'Moved %s 1-amp files'%counter
+
+
+counter = 0
+for filename in headers.keys():
+    if headers[filename]['NAMPSYX']=='1 2':
+        SafeMove(filename, dest_root_path + 'two_amp/' + GetFilename(filename))
+        counter += 1
+        headers.pop(filename)
+print 'Moved %s 2-amp files'%counter
+
+
+for filename in headers.keys():
+    date_obs = headers[filename]['DATE-OBS']
+    try: # see if compliant. Don't use, just a test with dafBase
+        dt = dafBase.DateTime(date_obs, dafBase.DateTime.TAI)
+    except: #if bad, sanitise
+        year, month, day, h, m, s = re.split(r"[-:T]", date_obs)
+        if re.search(r"[A-Z]$", s):
+            s, TZ = s[:-1], s[-1]
+        else:
+            TZ = ""
+
+        date_obs = "%4d-%02d-%02dT%02d:%02d:%06.3f%s" % (int(year), int(month), int(day),
+                                                         int(h),    int(m),     float(s), TZ)
```

Comment:
If you are going to do sanitization without writing it to the headers, I'd much prefer that this code be put in a free function in the library path and called from there rather than having it in two places.